File: ruby/easy/longest_palindrome.rb
require 'set'
# @param {String} s
# @return {Integer}
def longest_palindrome(s)
    len = 0
    set = Set.new
    s.each_char do |c|
        if set.include?(c)
            len += 2 and set.delete(c)
        else
            set.add(c)
        end
    end
    return len + ((set.length > 0) ? 1 : 0)
end

File: javascript/easy/longestPalindrome.js
/**
 * @param {string} s
 * @return {number}
 */
var longestPalindrome = function(s) {
    let map = new Map(), len = 0;
    for(let c of s) {
        let count = map.get(c);
        if(count === undefined) {
            map.set(c, 1);
        } else {
            len += 2;
            map.delete(c);
        }
    }
    return len + ((map.size > 0) ? 1 : 0);
};
